### DISCUSSION TITLE
TextLayer Disappearing

### DISCUSSION THREAD

**Discussion by ilan-gold** [visgl/deck.gl#5696] [Q&A] [answered]
Hello, 

Offhand, can anyone think of a reason why each `TextLayer` is disappearing (in the below gif) but not each `LineLayer` near which it lies (it's little hard to see, sorry)?  It doesn't appear to be an opacity issue with the volume since each `LineLayer` is visible.  I'm going to start trying to create a "reproducible example" later that doesn't involve volume ray casting but figured I'd ask in the meantime (while our [code](https://github.com/hms-dbmi/viv/compare/ilan-gold/new_avivator...ilan-gold/3d_axis?expand=1#diff-3fc0db66cbbb3aeed1ef52cfc32e4d7c22073300551f47a205e77ca90dc6df6b) is open-source so "reproducible" in some sense, there's a lot going on so no expectation for anyone to debug for me).

Thanks!!!!!
![ezgif com-gif-maker](https://user-images.githubusercontent.com/43999641/115930203-0e106e80-a457-11eb-89e9-a851516bd5ee.gif)

**Comment by sclebal** [comment]
Hi @ilan-gold, it's hard to know why the text layers are disappearing without a simple reproduction, but I'd advice you to store deck.gl instance as a global variable using `window` and access the layer array from the browser console: `deckInstance.props.layers`. There you can find you TextLayer and see the `data` property and inspect if the text is missing or the position is wrong. Might be the layers order is not correct, but I don't think so.

Hope it helps.

**Reply by ilan-gold** [reply]
@Clebal I've done some digging using the `window` and other means.  It appears that the issue is a general one with `TextLayer` - for example if I never remove the loading text, it also disappears nonetheless.  Another thing I'm noticing is that the scale bar lines are becoming dimmer even when I render the volume with `color = vec4(0.0);` at all coordinates.

**Reply by ilan-gold** [reply]
I should also mention that all data appears where it should be.

**Comment by ilan-gold** [comment, marked as answer]
The issue was with my setting
```javascript
 [GL.CULL_FACE]: true,
[GL.CULL_FACE_MODE]: GL.FRONT
```
which was being set globally.  I need this I think only to be run locally per (the volume) layer.
